src/solver/newton/newton.hpp

```diff
@@ -73,13 +71,17 @@ EIGEN_DEVICE_FUNC int newton_solver(Eigen::Matrix<double, N, 1>& X,
             // we have converged: return iteration count
             return iter;
         }
-#if defined(CORENEURON_ENABLE_GPU) && !defined(DISABLE_OPENACC)
-        X -= partialPivLu<N>(J, F);
-#else
-        // update X use in-place LU decomposition of J with partial pivoting
-        // (suitable for any N, but less efficient than .inverse() for N <=4)
-        X -= Eigen::PartialPivLU<Eigen::Ref<Eigen::Matrix<double, N, N>>>(J).solve(F);
-#endif
+        // In Eigen the default storage order is ColMajor.
+        // Crout's implementation requires matrices stored in RowMajor order (C-style arrays).
+        // Therefore, the transposeInPlace is critical such that the data() method to give the rows
+        // instead of the columns.
+        if (!J.IsRowMajor)
+            J.transposeInPlace();
```

Comment:
to clarify - when / how many times this transpose gets called?

Reply:
essentially every time, since the default storage order in Eigen is ColMajor.

Reply:
As we generate the code then couldn't we consistently use RowMajor when the Crout solver is (going to be) used?

Reply:
for now I would leave Eigen in its default state. Once we decide to completely remove Eigen, then we make this change. Performance-wise, Crout-branch even with this transposeInPlace is consistently faster compared to Eigen.